apps/evm/src/connect-ui/component/ConnectModal/EvmWalletList.tsx

```diff
@@ -0,0 +1,110 @@
+import { Avatar, Flex, List, ListItem, ListProps, Spinner } from '@gobob/ui';
+import { Connector } from '@gobob/wagmi';
+import { t } from '@lingui/macro';
+import { useLingui } from '@lingui/react';
+
+import { WalletIcon } from '../WalletIcon';
+
+enum WalletId {
+  OkexWallet = 'com.okex.wallet',
+  CoinbaseWalletSDK = 'coinbaseWalletSDK',
+  CoinbaseWallet = 'com.coinbase.wallet',
+  MetaMaskMobile = 'io.metamask'
+}
+
+type PlaceholderConnector = {
+  id: string;
+  name: string;
+  href: string;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  icon?: any;
+};
+
+const okxPlaceholder: PlaceholderConnector = {
+  id: WalletId.OkexWallet,
+  name: 'OKX Wallet',
+  href: 'https://www.okx.com/web3'
+};
+
+type Props = {
+  connector?: Connector;
+  connectors: readonly Connector[];
+  pendingConnector?: Connector;
+};
+
+type InheritAttrs = Omit<ListProps, 'children'>;
+
+type EvmWalletListProps = Props & InheritAttrs;
+
+const EvmWalletList = ({
+  onSelectionChange,
+  pendingConnector,
+  connectors: connectorsProp,
+  connector,
+  ...props
+}: EvmWalletListProps) => {
+  const { i18n } = useLingui();
+
+  const okxConnector = connectorsProp.find((c) => c.id === WalletId.OkexWallet) || okxPlaceholder;
+
+  const wagmiConnectors = connectorsProp.filter(
+    (c) =>
+      !(
+        (c.id === WalletId.CoinbaseWalletSDK && connectorsProp.some((el) => el.id === WalletId.CoinbaseWallet)) ||
+        c.id === okxConnector.id
+      )
+  );
```

Comment:
```suggestion
  const hasCOinbaseWallet = connectorsProp.some((el) => el.id === WalletId.CoinbaseWallet);
  const wagmiConnectors = connectorsProp.filter(
    (c) =>
      !(
        (c.id === WalletId.CoinbaseWalletSDK && hasCoinbaseWallet) ||
        c.id === okxConnector.id
      )
  );
```

it's not quite n^2 time complexity computation because connector list is of fixed size but still better to reduce amount of nested loops

Reply:
Some is not reached if the first clause is not true